lib/backend/watchersyncer/watchercache.go

```diff
@@ -432,13 +410,10 @@ func (wc *watcherCache) markAsValid(resourceKey string) {
 	}
 }
 
-// onError signals to the syncer that this watcherCache is not in-sync if the number of consecutive errors
-// exceeds the error threshold.  See finishResync() for how the watcherCache goes back to in-sync.
+// onError signals to the syncer that this watcherCache is not in-sync, hence perform resync.
+// See finishResync() for how the watcherCache goes back to in-sync.
 func (wc *watcherCache) onError() {
```

Comment:
I think we can either remove, or rename/repurpose this method.

We should be sending WaitForDatastore only if hasSynced is set to "true".  Otherwise we could send multiple WaitForDatastore events.  The owning watchersyncer counts WaitForDatastore/InSync events and they should balance out.

Perhaps:
-  Rename this to be "onWaitForDatastore()" method that sends a WaitForDatastore if not in-sync and then sets the in-sync flag to false.
-  Update the other place where we send WaitForDatastore to call this method instead.

Reply:
Update PR as per the above suggestion.